flink-connector-mysql-cdc/src/main/java/com/alibaba/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReader.java

```diff
@@ -32,36 +32,35 @@
 import com.alibaba.ververica.cdc.connectors.mysql.source.events.EnumeratorRequestReportEvent;
 import com.alibaba.ververica.cdc.connectors.mysql.source.events.SourceReaderReportEvent;
 import com.alibaba.ververica.cdc.connectors.mysql.source.offset.BinlogOffset;
+import com.alibaba.ververica.cdc.connectors.mysql.source.split.MySqlBinlogSplitState;
+import com.alibaba.ververica.cdc.connectors.mysql.source.split.MySqlSnapshotSplitState;
 import com.alibaba.ververica.cdc.connectors.mysql.source.split.MySqlSplit;
-import com.alibaba.ververica.cdc.connectors.mysql.source.split.MySqlSplitKind;
 import com.alibaba.ververica.cdc.connectors.mysql.source.split.MySqlSplitReader;
 import com.alibaba.ververica.cdc.connectors.mysql.source.split.MySqlSplitState;
 import org.apache.kafka.connect.source.SourceRecord;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 /** The source reader for MySQL source splits. */
-public class MySqlSourceReader<T>
+public class MySqlSourceReader<T, SplitT extends MySqlSplit>
```

Comment:
We don't need generic type `SplitT`, just use `MySqlSplit` is fine and will be much simpler.

Reply:
I think we need this because `MySqlSplitState` is parameterized generic type `MySqlSplitState<SplitT extends MySqlSplit>`